data_model/src/permissions.rs

```diff
@@ -11,19 +11,57 @@ use alloc::{
 #[cfg(feature = "std")]
 use std::collections::{btree_map, btree_set};
 
-use derive_more::Display;
+use derive_more::{Constructor, Display, FromStr};
 use getset::Getters;
 #[cfg(feature = "ffi")]
 use iroha_ffi::{ffi_export, IntoFfi, TryFromFfi};
 use iroha_schema::IntoSchema;
 use parity_scale_codec::{Decode, Encode};
 use serde::{Deserialize, Serialize};
 
-use crate::{Name, Value};
+use crate::{Identifiable, Name, Registered, Value};
 
 /// Collection of [`PermissionToken`]s
 pub type Permissions = btree_set::BTreeSet<PermissionToken>;
 
+/// Identification of a permission token.
+#[derive(
+    Debug,
+    Display,
+    Constructor,
+    FromStr,
+    Clone,
+    PartialEq,
+    Eq,
+    PartialOrd,
+    Ord,
+    Hash,
+    Decode,
+    Encode,
+    Deserialize,
+    Serialize,
+    IntoSchema,
+)]
+#[cfg_attr(feature = "ffi", derive(IntoFfi, TryFromFfi))]
+#[cfg_attr(feature = "ffi", ffi_export)]
+#[display(fmt = "{name}")]
+pub struct Id {
+    /// Permission token name
+    pub name: Name,
+}
+
+impl Identifiable for Id {
```

Comment:
I don't see other `Id`s implementing `Identifiable`? This seems redundant

Reply:
It's not only redundant, something is identifiable if it **has** an Id. This **is** an Id.

Reply:
Right, it's an awkward construction. `PermissionTokenDefinition` will fix it.